### ISSUE TITLE
Using Filters as factor levels for Mann-Whitney

### ISSUE THREAD

**Issue by ibinsnur** [jasp-stats/jasp-issues#633]
Hi, 

When doing a Mann-Whitney Independent Samples Test, I obviously can't use grouping variables with more than two factors. Unfortunately JASP also does not recognise filtering in this case. Thus you can't use a variable with several groups/factors as the grouping variable. In my case, there is a variable with several age groups and I'm only interested in comparing the youngest against the oldest group.  
Would be nice, if this could be fixed. 

Thank you, and keep up the good work!

**Comment by FransMeerhoff** [comment]
Hi @ibinsnur 
Thanks for reporting this.
Please can you send us some more information about OS, JASP version etc. as in the template is asked?

@vandenman  
Hi Don, can you have a look one (re-assign it)?

**Comment by vandenman** [comment]
@JorisGoosen After a discussion with @TimKDJ, we decided that it's probably best that when the dataset is loaded in R, any levels that are filtered out do not appear when calling `levels(variable)`. Do you agree/ can you make that happen?

**Comment by AlexanderLyNL** [comment]
@JorisGoosen didn't you do this already? I remember that you added a "drop=TRUE" in read factors, when we implemented the filter?

**Comment by JorisGoosen** [comment]
Yeah, that is supposed to happen in `jaspRCPP_makeFactor`, where first the variable is given all levels and then at the end `droplevels` is called on it. And this code has been in there for quite a while.

See line 956 of jasprcpp.cpp.

So maybe there is something wrong with the analysis instead?

**Comment by JorisGoosen** [comment]
@vandenman did you check if the levels are actually in the column?

**Comment by ibinsnur** [comment]
@FransMeerhoff I'm won't be at the computer today, where the problem occurred. But it's Windows 10 and JASP was installed two maybe three weeks ago, so I guess it's the latest 64bit version. 
Thank you guys for addressing it so fast!

**Comment by JorisGoosen** [comment]
Hmm, if I look at the cpp code then I do see a border-condition where something might go different, the end is:

```c++
	if(v.size() == 0)
		return v;

	static Rcpp::Function droplevels("droplevels");
	return droplevels(Rcpp::_["x"] = v);
```

So if the column has no values it gets all levels. Because otherwise it might not get levels at all? Although I assume that in @ibinsnur 's case there will still be data.

@ibinsnur could you perhaps upload your jasp-file, where this problem occurs? If the data isn't sensitive of course.

**Comment by ibinsnur** [comment]
@JorisGoosen I'm very sorry, but it actually is sensitive data. What I can do (tomorrow) is post screenshots without labels and give a more precise description of the variables used.

**Comment by JorisGoosen** [comment]
Well, maybe you could make a tiny example that has the same problem? It doesn't really need to be the actual data or sensible data even, just something that gives me a working example of the problem.

**Comment by ibinsnur** [comment]
@JorisGoosen Sounds good, will do. Please be patient as I maybe will need until Thursday or Friday.

**Comment by JorisGoosen** [comment]
Don't worry about it, we will be more patient than that!

**Comment by ibinsnur** [comment]
I found the problem: 
**TL;DR is at the bottom**

First attempt: Exported Data to csv -> edited in Excel (Deleted all other variables, renamed relevant variables) -> opened in JASP again -> JASP recognizes filtering for grouping variable

Tried it with the original Dataset (who knows...) --> JASP does not recognize filtering for the grouping variable

Second attempt: Exported Data to csv --> edited in Excel (just deleted the most sensible variables, renamed all other variables) --> JASP crashed everytime when I tried to open it. 
I edited the data a lot. Still crashed. When I deleted all the variables which had more in them than just a number (dates with dots, underscores, additional characters) JASP could open it. And more important...it recognzied filtering. 

Third attempt: Exported Data to csv --> systematically deleted and renamed variables --> After special characters were gone (string variables with "Ä", "Ö", etc. since it's german) JASP still could'nt open it. When dates were gone, JASP could open it and recognize the filter variable. (This might be a bug as well?) 

After all the export/edit/import procedure the labels were gone as well, so I added them again and found the problem (because it suddenly stopped working again):

**TL;DR**
If there is a special character used within the label (In my case it was an "Ü"), the problem (not recognizing filters) occurs. After changing it back from "Ü" to "UE", it worked again. 

:)

**Comment by AlexanderLyNL** [comment]
@JorisGoosen one of these windows UTF-8 encoding things again?

**Comment by JorisGoosen** [comment]
@ibisnur that is some good debugging there!

So, you mention "export to csv", from which application are you exporting? Just curious.
The fact that the problem recurs after adding the label says enough here, apparently there is something going wrong there. 

@AlexanderLyNL thought that this might be a Windows UTF-8 thing is likely close to the mark, I just spend some time looking at related problems (for translating) and I noticed that having characters with umlaut couldn't actually be exported properly from R to the UI on Windows only. This might be the inverse problem!

**Comment by ibinsnur** [comment]
@JorisGoosen I exported the .csv file directly form JASP. Before that I used SPSS for data preparation and consolidation, imported the .sav file into JASP (most labels got lost in the process btw) and saved it as a JASP file so I can make changes.

**Comment by JorisGoosen** [comment]
@ibinsnur there have been quite some improvements to reading SPSS data files in the upcoming version. I had to buy it though, because it can apparently throw quite a lot of variation in (weird) data formats/columns as it turns out.

I am working on this issue at the moment actually and it is easily reproducible. Hopefully it is as easy to fix.

When that is done I will ask you to test the new version and I hope you would like to help us out with it. The best would then be that you try to import your SPSS file directly and see if it now works out. 

And perhaps, you could send me a simple SPSS file with some umlauts or such in the labels of an ordinal column. Because I am working from home (and with the whole Corona-virus thing this will probably continue for a while) and I do not (yet) have SPSS available here. It would be good to verify that that import goes right. Because that is another variation of data that I think I missed up until now.
